code/src/main/java/com/onlinecontacttracing/messaging/MessageServlet.java

```diff
@@ -0,0 +1,97 @@
+import com.onlinecontacttracing.messaging.filters.CheckMessagesForFlags;
+import java.util.List;
+
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
+import java.util.ArrayList;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.util.Collections;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.Projection;
+
+@WebServlet("/messaging")
+public class MessageServlet extends HttpServlet {
+  private SystemMessage systemMessage;
+  private LocalityResource localityResource;
+  private CustomizableMessage customizableMessage;
+  private String userMessage;
+  private PositiveUser user;
+  //for backend use
+  private String compiledMessage;
+  //for frontend display
+  private List<String> statusMessage = new ArrayList<String> ();
+
+
+  public MessageServlet(SystemMessage systemMessage, LocalityResource localityResource, CustomizableMessage customizableMessage, PositiveUser user) {
+    this.systemMessage = systemMessage;
+    this.localityResource = localityResource;
+    this.customizableMessage =  customizableMessage;
+    this.userMessage = customizableMessage.getMessage();
+    this.user = user;
+
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html;");
+    String messageLanguage = request.getParameter("language");
+    response.getWriter().println(statusListToShowUser(messageLanguage));
+  }
+
+  public void checkForFlags() {
+    CheckMessagesForFlags flagChecker = new CheckMessagesForFlags();
+    statusMessage = CheckMessagesForFlags.findTriggeredFlags(flagChecker, user, userMessage);
+  }
+
+  //if flags are triggered, returns message and list of errors, else returns message
+  public List<String> statusListToShowUser(String messageLanguage) {
+    String translatedResourceMessage;
+    String translatedSystemMessage;
+    user.incrementAttemptedEmailDrafts();
+    checkForFlags();
+    
+    if (messageLanguage.equals("SP")) {
+    translatedResourceMessage = localityResource.getEnglishTranslation();
+    translatedSystemMessage = systemMessage.getEnglishTranslation();
+    }
+    else {
+      translatedResourceMessage = localityResource.getEnglishTranslation();
+      translatedSystemMessage = systemMessage.getEnglishTranslation();
+    }
+    
+    //if any flags are triggered, do not want to include the user's message in the message sent to the backend
+    if (statusMessage.size() != 0){
```

Comment:
statusMessage.size() > 0 

makes more explicit what you intend for this here.

Reply:
Also, what is the userMessage in the case that statusMessage is size 0? If you can make that be equivalent to an empty string, then you can just concat without this check and clean things up slightly.